loadgen/generic_executor.go

```diff
@@ -106,24 +127,30 @@ func (g *genericRun) Run(ctx context.Context) error {
 		go func() {
 			startTime := time.Now()
 			err := g.executor.Execute(ctx, run)
-			// Only log/wrap/send to channel if context is not done
-			if ctx.Err() == nil {
-				if err != nil {
-					err = fmt.Errorf("iteration %v failed: %w", run.Iteration, err)
-					g.logger.Error(err)
-				}
-				select {
-				case <-ctx.Done():
-				case doneCh <- err:
-					// Record/log here, not if it was cut short by context complete
-					g.executeTimer.Record(time.Since(startTime))
-				}
+			if err != nil {
+				err = fmt.Errorf("iteration %v failed: %w", run.Iteration, err)
+				g.logger.Error(err)
+			}
+			select {
+			case <-ctx.Done():
+			case doneCh <- err:
+				// Record/log here, not if it was cut short by context complete
+				g.executeTimer.Record(time.Since(startTime))
 			}
 		}()
 	}
-	// Wait for all to be done or an error to occur
-	for runErr == nil && ctx.Err() == nil && currentlyRunning > 0 {
-		waitOne()
+	g.logger.Debugf("No longer starting new iterations")
+	if currentlyRunning == 0 {
+		return fmt.Errorf("no runs were started. Iterations should be >0 or duration should be >0")
```

Comment:
> "no runs were started"

`currentlyRunning` could also be zero at this point due to an error setting `runErr`, right?

Reply:
Yeah, good point.